mopidy_musicbox_webclient/static/js/controls.js

```diff
@@ -604,6 +613,36 @@ function showFavourites() {
     });    
 }
 
+function upgradeStreamUrisToFavourites() {
+    $.cookie.json = true;
+    var streamUris = $.cookie('streamUris'); // Read the cookie.
+    if (streamUris) {
+        var uris = []; // Prepare a list of uris to lookup.
+        for (var key in streamUris) {
+            var rs = streamUris[key];
+            if (rs) {
+                uris.push(rs[1]);
+            }
+        }
+        mopidy.library.lookup({'uris': uris}).then(function(results) {
+            var tracks = []; // Prepare a list of tracks to add.
+            for (var key in streamUris) {
+                var rs = streamUris[key];
+                if (rs) {
+                    var track = results[rs[1]][0];
+                    track.name = rs[0] || track.name; // Use custom name if provided.
```

Comment:
This fails for me if ```mopidy.library.lookup()``` could not retrieve a track for the given URI. And the exception causes the entire loop to abort and not do ```addToFavourites(tracks);``` for any of the streams in the cookie (so nothing is added to the favourites playlist).

The easiest way to replicate this is to save a geo-blocked youtube stream to the cookie and then try to upgrade (in my case that is one of the standard ones distributed in ```streamuris.js```: http://www.youtube.com/watch?v=eQsM6u0a038